File: Cpp/example2.cpp
#include "pbPlots.hpp"
#include "supportLib.hpp"

using namespace std;

int main(){
	bool success;
	StringReference *errorMessage = CreateStringReferenceLengthValue(0, L' ');
	RGBABitmapImageReference *imageReference = CreateRGBABitmapImageReference();

	vector<double> xs{-2, -1, 0, 1, 2};
	vector<double> ys{2, -1, -2, -1, 2};

	ScatterPlotSeries *series = GetDefaultScatterPlotSeriesSettings();
	series->xs = &xs;
	series->ys = &ys;
	series->linearInterpolation = true;
	series->lineType = toVector(L"dashed");
	series->lineThickness = 2;
	series->color = GetGray(0.3);

	ScatterPlotSettings *settings = GetDefaultScatterPlotSettings();
	settings->width = 600;
	settings->height = 400;
	settings->autoBoundaries = true;
	settings->autoPadding = true;
	settings->title = toVector(L"x^2 - 2");
	settings->xLabel = toVector(L"X axis");
	settings->yLabel = toVector(L"Y axis");
	settings->scatterPlotSeries->push_back(series);

	success = DrawScatterPlotFromSettings(imageReference, settings, errorMessage);

	if(success){
		vector<double> *pngdata = ConvertToPNG(imageReference->image);
		WriteToFile(pngdata, "example2.png");
		DeleteImage(imageReference->image);
	}else{
		cerr << "Error: ";
		for(wchar_t c : *errorMessage->string){
			wcerr << c;
		}
		cerr << endl;
	}

	FreeAllocations();

	return success ? 0 : 1;
}

File: JavaScript/example2.js
var fs = require('fs');

eval(fs.readFileSync('pbPlots.js')+'');
eval(fs.readFileSync('supportLib.js')+'');

var imageReference = CreateRGBABitmapImageReference();

var series = GetDefaultScatterPlotSeriesSettings();
series.xs = [-2, -1, 0, 1, 2];
series.ys = [2, -1, -2, -1, 2];
series.linearInterpolation = true;
series.lineType = "dashed";
series.lineThickness = 2;
series.color = GetGray(0.3);

var settings = GetDefaultScatterPlotSettings();
settings.width = 800;
settings.height = 600;
settings.autoBoundaries = true;
settings.autoPadding = true;
settings.title = "x^2 - 2";
settings.xLabel = "X axis";
settings.yLabel = "Y axis";
settings.scatterPlotSeries = [series];

DrawScatterPlotFromSettings(imageReference, settings);

var pngdata = ConvertToPNG(imageReference.image);
WriteToFile(pngdata, "example2.png");
DeleteImage(imageReference.image);
